baseapp/state.go

```diff
@@ -2,11 +2,13 @@ package baseapp
 
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	abci "github.com/tendermint/tendermint/abci/types"
 )
 
 type state struct {
-	ms  sdk.CacheMultiStore
-	ctx sdk.Context
+	ms           sdk.CacheMultiStore
+	ctx          sdk.Context
+	eventHistory []abci.Event
```

Comment:
is the extra field here truly necessary? im curious if this PR can work by just storing the events in the state's existing ctx event manager.